# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1338223203 25200
#      Mon May 28 09:40:03 2012 -0700
# Node ID c2411bff11c6339a3660e7102ed468140cfc7867
# Parent  775fed9ed47dfd426127da71f5744f9fcc0d3a07
test: Temporary patch to stop 'mkdir' tests failing until 'mkdir -p' is corrected.

* test_system.m: Save and restore pwd when running mkdir tests.

diff --git a/test/test_system.m b/test/test_system.m
--- a/test/test_system.m
+++ b/test/test_system.m
@@ -119,21 +119,29 @@
 %% test/octave.test/system/readdir-2.m
 %!error <Invalid call to readdir> readdir ()
 
 %% test/octave.test/system/readdir-3.m
 %!error <Invalid call to readdir> readdir ("foo", 1)
 
 %% test/octave.test/system/mk-rm-dir-1.m
 %!test
+%! ## FIXME: saving and restoring of pwd in olldir is a hack
+%! ##        'mkdir' should not change pwd but it does since
+%! ##        changeset 14679:a543ed02e673
+%! ##        which created 'mkdir -p' capabilities.  
+%! ##        When 'mkdir' has been corrected, delete this FIXME
+%! ##        and any lines with 'HACK'.
+%! olddir = pwd;   # HACK Line #1
 %! nm = tmpnam ();
 %! e1 = mkdir (nm);
 %! [s2, e2] = stat (nm);
 %! e3 = rmdir (nm);
 %! [s4, e4] = stat (nm);
+%! cd (olddir);    # HACK Line #2
 %! assert ((e1 && strcmp (s2.modestr(1), "d") && e3 && e4 < 0));
 
 %% test/octave.test/system/mkdir-1.m
 %!error <Invalid call to mkdir> mkdir ()
 
 %% test/octave.test/system/mkdir-2.m
 %!error <Invalid call to mkdir> mkdir ("foo", 1, 2)
 
